descqa/NumberDensityVersusRedshift.py

```diff
@@ -0,0 +1,307 @@
+                validation_data['mag_hi'] = []
+                validation_data['mag_lo'] = [float(m) for m in range(int(mag_hi_lim),int(mag_lo_lim)+1)]
+                validation_data['label'] = DEEP2_JAN_lbl
+            else:
+                raise ValueError('Observation {} not available'.format(observation))
+        else:
+            raise ValueError("{}-band data file {} not found or is empty".format(band,filename))
+            
+        return validation_data
+
+    def run_on_single_catalog(self, galaxy_catalog, catalog_name, base_output_dir):
+        
+        #check for skip_messages
+        if(hasattr(self,'skip_message')):
+            return TestResult(skipped=True, summary= self.skip_message)
+
+        #get catalog data
+        mag_field = galaxy_catalog.first_available(*self.possible_mag_fields)
+        if not mag_field:
+            return TestResult(skipped = True,summary = 'Missing requested quantities')
+        catalog_data = self.get_catalog_data(galaxy_catalog, [self.z, mag_field], filters=self.filters)
+        filtername = mag_field.partition(self.band + '_')[-1].upper()
+        filelabel = '_'.join([filtername,self.band])
+
+        fig, ax  =  plt.subplots(self.nrows,self.ncolumns,figsize=(figx_p,figy_p),sharex='col')
+        self.yaxis = 'P(z|m)' if self.normed else 'N(z|m)'
+
+        catalog_color = next(self.colors)
+        
+        #loop over magnitude cuts and make plots
+        results = {}
+        for n,cut in enumerate(self.mag_lo):
+            #initialize default values to use for validation-data fits
+            z0 = 0.
+            z0err = 0.
+            #get masks for magnitude cuts and get validation-data fits depending on selected observation
+            if self.observation.find(Coil2004)!= -1:
+                mask = (catalog_data[mag_field]<=cut) & (catalog_data[mag_field]>self.mag_hi[n])
+                cutlabel = ' '.join(['$',str(self.mag_hi[n]),'< $',self.band,'$<=',str(cut),'$'])
+                
+                #find correct row in validation data fits to use in plot
+                omask = (self.mag_hi.astype(int)==int(self.mag_hi[n])) & (self.mag_lo.astype(int)==int(cut))
+                z0 = self.validation_data['z0values'][omask]
+                if('z0errors' in self.validation_data.keys()):
+                    z0err = self.validation_data['z0errors'][omask]
+            else:
+                mask = (catalog_data[mag_field]<cut)
+                cutlabel =  ' '.join([self.band,'$<',str(cut),'$'])
+                if(self.observation.find(DEEP2_JAN)!= -1):
+                    z0 = self.validation_data['z0const'] + self.validation_data['z0linear']*(cut)
+            total = '(# of galaxies = {})'.format(np.sum(mask))
+ 
+            #bin catalog_data
+            N,binEdges = np.histogram(catalog_data[self.z][mask],bins=self.zbins)
+            sumz,binEdges = np.histogram(catalog_data[self.z][mask],bins=self.zbins,weights=catalog_data[self.z][mask])
```

Comment:
Replace all `catalog_data[self.z][mask]` to `z_this`

one space after comma

Reply:
Done